Userland/Applications/HexEditor/HexEditor.cpp

```diff
@@ -546,31 +569,60 @@ void HexEditor::paint_event(GUI::PaintEvent& event)
                 (character_width() * 3),
                 line_height() - m_line_spacing
             };
+
+            Gfx::Color background_color = palette().color(background_role());
+            Gfx::Color text_color = edited_flag ? Color::Red : palette().color(foreground_role());
+
             if (highlight_flag) {
-                painter.fill_rect(hex_display_rect, palette().selection());
-                text_color = text_color == Color::Red ? Color::from_rgb(0xFFC0CB) : palette().selection_text();
-            } else if (byte_position == m_position) {
-                painter.fill_rect(hex_display_rect, palette().inactive_selection());
+                background_color = palette().selection();
+                text_color = edited_flag ? Color::from_rgb(0xFFC0CB) : palette().selection_text();
+            } else if (byte_position == m_position && m_edit_mode == EditMode::Text) {
+                background_color = palette().inactive_selection();
                 text_color = palette().inactive_selection_text();
             }
+            painter.fill_rect(hex_display_rect, background_color);
 
             auto line = String::formatted("{:02X}", m_buffer[byte_position]);
             painter.draw_text(hex_display_rect, line, Gfx::TextAlignment::TopLeft, text_color);
 
+            if (m_edit_mode == EditMode::Hex) {
+                if (byte_position == m_position && m_cursor_blink_active) {
+                    Gfx::IntRect cursor_position_rect {
+                        hex_display_rect.left() + m_byte_position * character_width(), hex_display_rect.top(), 2, hex_display_rect.height() 
+                    };
+                    painter.fill_rect(cursor_position_rect, palette().text_cursor());
+                }
+            }
+
             Gfx::IntRect text_display_rect {
                 frame_thickness() + offset_margin_width() + (bytes_per_row() * (character_width() * 3)) + (j * character_width()) + 20,
                 frame_thickness() + 5 + (i * line_height()),
                 character_width(),
                 line_height() - m_line_spacing
             };
-            // selection highlighting.
+
+            background_color = palette().color(background_role());
+            text_color = edited_flag ? Color::Red : palette().color(foreground_role());
+            
             if (highlight_flag) {
-                painter.fill_rect(text_display_rect, palette().selection());
-            } else if (byte_position == m_position) {
-                painter.fill_rect(text_display_rect, palette().inactive_selection());
+                background_color = palette().selection();
+                text_color = edited_flag ? Color::from_rgb(0xFFC0CB) : palette().selection_text();
```

Comment:
Why a magic hex color here? I think a palette color or a predefined color would be better.

Reply:
Definitely. I figured I just change as little as possible. There does not seem to be a palette color that fits the purpose here so for now changed selected bytes are rendered with inverted colors.